modules/aruco/src/aruco.cpp

```diff
@@ -59,64 +59,18 @@ int estimatePoseBoard(InputArrayOfArrays corners, InputArray ids, const Ptr<Boar
     return (int)objPoints.total() / 4;
 }
 
-
-/**
-  * Check if a set of 3d points are enough for calibration. Z coordinate is ignored.
-  * Only axis parallel lines are considered
-  */
-static bool _arePointsEnoughForPoseEstimation(const vector<Point3f> &points) {
-    if(points.size() < 4) return false;
-
-    vector<double> sameXValue; // different x values in points
-    vector<int> sameXCounter;  // number of points with the x value in sameXValue
-    for(unsigned int i = 0; i < points.size(); i++) {
-        bool found = false;
-        for(unsigned int j = 0; j < sameXValue.size(); j++) {
-            if(sameXValue[j] == points[i].x) {
-                found = true;
-                sameXCounter[j]++;
-            }
-        }
-        if(!found) {
-            sameXValue.push_back(points[i].x);
-            sameXCounter.push_back(1);
-        }
-    }
-
-    // count how many x values has more than 2 points
-    int moreThan2 = 0;
-    for(unsigned int i = 0; i < sameXCounter.size(); i++) {
-        if(sameXCounter[i] >= 2) moreThan2++;
-    }
-
-    // if we have more than 1 two xvalues with more than 2 points, calibration is ok
-    if(moreThan2 > 1)
-        return true;
-    return false;
-}
-
 bool estimatePoseCharucoBoard(InputArray charucoCorners, InputArray charucoIds,
-                                           const Ptr<CharucoBoard> &board, InputArray cameraMatrix,
-                                           InputArray distCoeffs, InputOutputArray rvec,
-                                           InputOutputArray tvec, bool useExtrinsicGuess) {
+                              const Ptr<CharucoBoard> &board, InputArray cameraMatrix,
+                              InputArray distCoeffs, InputOutputArray rvec,
+                              InputOutputArray tvec, bool useExtrinsicGuess) {
     CV_Assert((charucoCorners.getMat().total() == charucoIds.getMat().total()));
 
-    // need, at least, 4 corners
-    if(charucoIds.getMat().total() < 4) return false;
-
-    vector<Point3f> objPoints;
-    objPoints.reserve(charucoIds.getMat().total());
-    for(unsigned int i = 0; i < charucoIds.getMat().total(); i++) {
-        int currId = charucoIds.getMat().at< int >(i);
-        CV_Assert(currId >= 0 && currId < (int)board->getChessboardCorners().size());
-        objPoints.push_back(board->getChessboardCorners()[currId]);
-    }
-
-    // points need to be in different lines, check if detected points are enough
-    if(!_arePointsEnoughForPoseEstimation(objPoints)) return false;
+    // get object and image points for the solvePnP function
+    Mat objPoints, imgPoints;
+    board->matchImagePoints(charucoCorners, charucoIds, objPoints, imgPoints);
+    solvePnP(objPoints, imgPoints, cameraMatrix, distCoeffs, rvec, tvec, useExtrinsicGuess);
```

Comment:
solvePnP throws exception, if there are no enough points.

Reply:
added points check:
`if(charucoIds.getMat().total() < 4) return false;`

Reply:
I propose to add try-catch block and return false in case of exception. The function did not throw exceptions before and should not do it after the PR.

Reply:
Add try catch:
```
    try {
        solvePnP(objPoints, imgPoints, cameraMatrix, distCoeffs, rvec, tvec, useExtrinsicGuess);
    }
    catch (const cv::Exception& e) {
        CV_LOG_WARNING(NULL, "estimatePoseCharucoBoard: " << std::endl << e.what());
        return false;
    }
```